cogs/birthday/core.py

```diff
@@ -0,0 +1,215 @@
+import asyncio
+from datetime import datetime, timedelta
+import logging
+import os
+from random import choice
+import time
+
+import discord
+
+from redbot.core import Config, data_manager
+from redbot.core.bot import Red
+
+from .constants import *
+
+
+class Core:
+    # Class constructor
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.bgTask: asyncio.Task = None
+        self.config: Config = None
+        self.lastChecked: datetime = None
+        self.logger: logging.Logger = None
+
+        self.initializeConfigAndLogger()
+        self.initializeBgTask()
+
+    def initializeConfigAndLogger(self):
+        self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
+        # Register default (empty) settings.
+        self.config.register_guild(**BASE_GUILD)
+        self.config.register_member(**BASE_GUILD_MEMBER)
+
+        # Initialize logger, and save to cog folder.
+        saveFolder = data_manager.cog_data_path(cog_instance=self)
+        self.logger = logging.getLogger("red.luicogs.Birthday")
+        if not self.logger.handlers:
+            logPath = os.path.join(saveFolder, "info.log")
+            handler = logging.FileHandler(filename=logPath, encoding="utf-8", mode="a")
+            handler.setFormatter(
+                logging.Formatter("%(asctime)s %(message)s", datefmt="[%d/%m/%Y %H:%M:%S]")
+            )
+            self.logger.addHandler(handler)
+
+    def initializeBgTask(self):
+        # On cog load, we want the loop to run once.
+        self.lastChecked = datetime.now() - timedelta(days=1)
+        self.bgTask = self.bot.loop.create_task(self.birthdayLoop())
+
+    # Cancel the background task on cog unload.
+    def __unload(self):  # pylint: disable=invalid-name
+        self.bgTask.cancel()
+
+    def cog_unload(self):
+        self.__unload()
```

Comment:
Can we remove the `__unload` method and just leave `cog_unload`? I think this was ported from cog v2 days where it was looking for `__unload()`, so I don't think it's needed any more.
```suggestion
    def cog_unload(self):
        self.bgTask.cancel()
```